cunumeric/fft/fft.py

```diff
@@ -0,0 +1,857 @@
+
+    See Also
+    --------
+    numpy.fft.irfftn
+
+    Availability
+    --------
+    GPU
+    """
+    # Convert to complex if real
+    if a.dtype == np.float32:
+        a = a.astype(np.complex64)
+    elif a.dtype == np.float64:
+        a = a.astype(np.complex128)
+    # Retrieve type
+    fft_type = FFTCode.complex_to_real_code(a.dtype)
+
+    s, axes = _sanitize_user_axes(a, s, axes, is_c2r=True)
+
+    # Operate by axes
+    if _operate_by_axes(a, axes):
+        if len(axes) > 1:
+            c2r = a.fft(s=s[0:-1], axes=axes[0:-1], kind=fft_type.complex, direction=FFTDirection.INVERSE, norm=norm)
+        else:
+            c2r = a
+        return c2r.fft(s=[s[-1]], axes=[axes[-1]], kind=fft_type, direction=FFTDirection.INVERSE, norm=norm)
+    # Operate as a single FFT
+    else:
+        # cuFFT out-of-place C2R always overwrites the input buffer, which is not what we want here, so copy
+        b = a.copy()
+        return b.fft(s=s, axes=axes, kind=fft_type, direction=FFTDirection.INVERSE, norm=norm)
+
+
+@add_boilerplate("a")
+def hfft(a, n=None, axis=-1, norm=None):
+    """
+    Compute the FFT of a signal that has Hermitian symmetry, i.e., a real
+    spectrum.
+
+    Parameters
+    ----------
+    a : array_like
+        The input array.
+    n : int, optional
+        Length of the transformed axis of the output. For `n` output
+        points, ``n//2 + 1`` input points are necessary.  If the input is
+        longer than this, it is cropped.  If it is shorter than this, it is
+        padded with zeros.  If `n` is not given, it is taken to be ``2*(m-1)``
+        where ``m`` is the length of the input along the axis specified by
+        `axis`.
+    axis : int, optional
+        Axis over which to compute the FFT. If not given, the last
+        axis is used.
+    norm : {"backward", "ortho", "forward"}, optional
+        Normalization mode (see `numpy.fft`). Default is "backward".
+        Indicates which direction of the forward/backward pair of transforms
+        is scaled and with what normalization factor.
+
+    Returns
+    -------
+    out : ndarray
+        The truncated or zero-padded input, transformed along the axis
+        indicated by `axis`, or the last one if `axis` is not specified.
+        The length of the transformed axis is `n`, or, if `n` is not given,
+        ``2*m - 2`` where ``m`` is the length of the transformed axis of
+        the input. To get an odd number of output points, `n` must be
+        specified, for instance as ``2*m - 1`` in the typical case,
+
+    Notes
+    ------
+
+    See also
+    --------
+    numpy.fft.hfft
+
+    Availability
+    --------
+    GPU
+    """
+    s = (n,) if n is not None else None
+    axis = (axis,) if axis is not None else None
+    # Add checks to ensure input is hermitian?
+    # Essentially a C2R FFT, with reverse sign (forward transform, forward norm)
+    return irfftn(a=a.conjugate(), s=s, axes=axis, norm=FFTNormalization.reverse(norm))
```

Comment:
Can we do `norm.reverse` ? Like `fft_type.complex` above.

Reply:
I thought about it, but the return value of reverse is not a norm, it's a string similar to the one provided by the user as input and I'd like to keep it as a method of the class more than a property of the norm.